test_socket.py

```diff
@@ -0,0 +1,232 @@
+import socket
+import string
+import email.utils
+
+
+def test_text():
+    ADDR = ('127.0.0.1', 10007)
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    client.connect(ADDR)
+    msg = "GET sample.txt HTTP/1.1"+"\r\n"+"Host: www.example.com"+"\r\n\r\n"
+    client.sendall(msg)
+    client.shutdown(socket.SHUT_WR)
+    part = ""
+    while True:
+        tempPart = client.recv(16)
+        part = part + tempPart
+        if len(tempPart) < 16:
+            client.close()
+            break
+    part_list = string.split(part, '\r\n')
+    assert part_list[0].find("200 OK") != -1
+    date = email.utils.formatdate(usegmt=True)
+    assert part_list[1].find(date) != -1
+    assert part_list[2].find("Content-Type: text/plain") != -1
+    assert part_list[3].find("Content-Length:") != -1
+    assert part_list[5].find("This is a very simple text file") != -1
+
+
+def test_directory():
+    ADDR = ('127.0.0.1', 10007)
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    client.connect(ADDR)
+    msg = "GET images HTTP/1.1"+"\r\n"+"Host: www.example.com"+"\r\n\r\n"
+    client.sendall(msg)
+    client.shutdown(socket.SHUT_WR)
+    part = ""
+    while True:
+        tempPart = client.recv(16)
+        part = part + tempPart
+        if len(tempPart) < 16:
+            client.close()
+            break
+    part_list = string.split(part, '\r\n')
+    assert part_list[0].find("200 OK") != -1
+    date = email.utils.formatdate(usegmt=True)
+    assert part_list[1].find(date) != -1
+    assert part_list[2].find("Content-Type: text/html") != -1
+    assert part_list[3].find("Content-Length:") != -1
+    assert part_list[5].find("JPEG_example.jpg") != -1
+
+
+def test_jpg():
+    ADDR = ('127.0.0.1', 10007)
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    client.connect(ADDR)
+    msg = "GET images/JPEG_example.jpg HTTP/1.1"+"\r\n"+"Host: www.example.com"+"\r\n\r\n"
+    client.sendall(msg)
+    client.shutdown(socket.SHUT_WR)
+    part = ""
+    while True:
+        tempPart = client.recv(16)
+        part = part + tempPart
+        if len(tempPart) < 16:
+            client.close()
+            break
+    part_list = string.split(part, '\r\n')
+    assert part_list[0].find("200 OK") != -1
+    date = email.utils.formatdate(usegmt=True)
+    assert part_list[1].find(date) != -1
+    assert part_list[2].find("Content-Type: image/jpeg") != -1
+    assert part_list[3].find("Content-Length:") != -1
```

Comment:
Could you decompose the content length header here, get the byte length and then validate that the actual body of this response is the length advertised by the header?  That'd be a useful test.